src/Workspaces/Core/Portable/Options/DocumentOptionSet.cs

```diff
@@ -45,6 +51,9 @@ public DocumentOptionSet WithChangedOption<T>(PerLanguageOption<T> option, T val
             return (DocumentOptionSet)WithChangedOption(option, _language, value);
         }
 
+        internal override AnalyzerConfigOptions CreateAnalyzerConfigOptions(IOptionService optionService, string? language)
+            => _backingOptionSet.AsAnalyzerConfigOptions(optionService, language ?? _language);
```

Comment:
Can `language` be different from `_language` here? If not, can we assert or throw an exception.

Reply:
I'm not sure what the right thing to do here is.

1. We could always use `_language`
2. We could always use `language`
3. We could prefer `language` but fall back to `_language` if it's not provided

Reply:
Seems like conflicts can only happen here when you end up with a document option set for a specific document (and language), but then invoke `OptionSet.AsAnalyzerConfigOptions(language)` with a different language. Assuming that is an unlikely scenario, I think `Debug.Assert(language == null || language == _language);` might hold here. I'd let you decide if having that assert in place could help identify if the above conflicting scenario is indeed a possibility. Your current implementation (3) seems reasonable to me.

Reply:
yes, i would like teh assert/runtime-check that these are cosistent.

Reply:
Now added